### PR TITLE
feat(primitives): add additional common aliases

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Closes: https://github.com/alloy-rs/core/issues/653

For context + related PR in `alloy`: https://github.com/alloy-rs/alloy/pull/859

## Solution

Add generic aliases unrelated to specific EIPs

- `TxNonce` (with doc alias)
- `BlockTimestamp`

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/primitives/src/aliases.rs [hunk 1]
```diff
@@ -65,6 +65,9 @@ pub type BlockHash = B256;
 /// A block number.
 pub type BlockNumber = u64;
 
+/// A block timestamp.
+pub type BlockTimestamp = u64;
```
Comment (onbjerg) [inline crates/primitives/src/aliases.rs:69]: unsure if we should add this w/o also using it, so maybe a separate pr?

**Review by onbjerg — APPROVED** [review]
lgtm otherwise, no strong feeling either way, just dont want us to end up not using the new alias:)

File: crates/primitives/src/aliases.rs (hunk 1 above)
Comment (zerosnacks) [reply crates/primitives/src/aliases.rs:69]: I would update https://github.com/alloy-rs/alloy/pull/859 to include it

Same for `TxNonce`

**Comment by zerosnacks** [comment]
> hmm, I'm sceptical that these aliases are useful

Added some additional context here: https://github.com/alloy-rs/alloy/pull/859

>Reduces magic values, reduces chance of error and makes it easier for end-users to see what value they are getting e.g. TxHash instead of FixedBytes<32> when watch()-ing a transaction. Main goal is to limit showing internal types and help with consistency / standardize fields (`u64` vs `U64`) and sizes.

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**